src/Material/Menu/Oracle.elm

```diff
@@ -0,0 +1,87 @@
+module Material.Menu.Oracle
+  ( Oracle, Element
+  , decode, decode'
+  ) where
+
+import DOM
+import Json.Decode exposing (..)
+
+{-| An Oracle stores relevant information from DOM during Toggle and Close
+events. This computes more than it needs to.
+-}
+
+type alias Oracle =
+  { button : Element
+  , menu : Element
+  , container : Element
+  , offsetTops : List Float
+  , offsetHeights : List Float
+  }
+
+type alias Element =
+  { offsetTop : Float
+  , offsetLeft : Float
+  , offsetHeight : Float
+  , bounds : DOM.Rectangle
+  }
+
+{-| Decode Oracle from the button's reference
+-}
+
+decode : Decoder Oracle
+decode =
+  object5
+    ( \button container menu offsetTops offsetHeights ->
+
+        { button = button
+        , menu   = menu
+        , container = container
+        , offsetTops = offsetTops
+        , offsetHeights = offsetHeights
+        }
+    )
+    ("target" := element)
+    (at ["target", "nextSibling"]  element)
```

Comment:
1. Is it possible to get rid of the repeated `(at ["target", "nextSibling"] ...)`, maybe by splitting this decoder into two?
2. Could you add `nextSibling` to the `DOM` module and submit a PR? In that case, this could become the slightly neater `(DOM.target (DOM.nextSibling (DOM.childNode 1 element)))`. 

(If only Elm had a right-associate `<|`.)